### PR TITLE
JENA-1550: TDB2 loader

### PR DESCRIPTION
**Pull request by afs**

### CODE REVIEW

File: jena-base/src/main/java/org/apache/jena/atlas/lib/Timer.java [hunk 1]
```diff
@@ -69,4 +69,13 @@ static public String timeStr(long timeInterval) {
     protected String timeStr(long timePoint, long startTimePoint) {
         return timeStr(timePoint - startTimePoint) ;
     }
+
+    /** Time an operation. Return the elapsed time in milliseconds. */
+    public static long time(Runnable action) {
```
Comment (rvesse) [inline jena-base/src/main/java/org/apache/jena/atlas/lib/Timer.java]: There's an implied assumption here that any exceptions thrown by the runnable are handled by the caller

File: jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderConst.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.jena.atlas.lib.tuple.Tuple;
+import org.apache.jena.tdb2.store.NodeId;
+
+public class LoaderConst {
+
+    /** Chunk size for the triple->tuples output pipe */  
+    public final static int ChunkSize = 100_000 ;
```
Comment (rvesse) [inline jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderConst.java]: Didn't even know this little syntax trick for making numeric constants more readable existed in Java!

File: jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import org.apache.jena.tdb2.sys.TDBInternal;
+
+/**
+ * The parallel Loader.
+ * <p>
+ * The process is:
+ * <blockquote>
+ * {@code DataBatcher -> DataToTuples -> Indexer}
+ * <blockquote>
+ * {@link DataBatcher} produces {@link DataBlock}s - grouping of triples and quads. It uses
```
Comment (rvesse) [inline jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java]: Unterminated sentence here?

File: jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+/**
+ * The parallel Loader.
+ * <p>
+ * The process is:
+ * <blockquote>
+ * {@code DataBatcher -> DataToTuples -> Indexer}
+ * <blockquote>
+ * {@link DataBatcher} produces {@link DataBlock}s - grouping of triples and quads. It uses 
+ * <br/>
+ * {@link DataToTuples} processes {@link DataBlock} to create 2 outputs blocks of {@code Tuple<NodeId>}, one output for triples, oue for quads.
```
Comment (rvesse) [inline jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java]: Typo `oue` -> `one`

File: jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java (hunk 3 above)
Comment (afs) [reply jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/loader/parallel/LoaderParallel.java]: Done.

Also some adding of "public" to what are internal classes to allow further experimentation (as time permits). Loading at larger scale, and to disk, benefits from phasing the index builds (c.f. tdbloader1) but it needs experimentation.

**Pull request merged**